scripts/run-test.ts

```diff
@@ -82,7 +82,7 @@ const main = async () => {
 						z.literal("DirectUpload").transform(() => Trigger.DirectUpload),
 					])
 					.default("GitHub"),
-				wranglerVersion: z.string().default("beta"),
+				installWrangler: z.union([z.null(), z.string()]).default(null),
```

Comment:
Great idea! But I think you want this instead:

```suggestion
				installWrangler: z.union([z.null().transform(() => "beta"), z.string(), z.undefined()]),
```

Reply:
yes sorry, my implementation was a bit off, thanks! 😓 🙏

Reply:
your suggested change seems to have some typing issues 😓 
![Screenshot 2023-08-11 at 16 58 49](https://github.com/GregBrimble/pages-e2e-tests/assets/61631103/97df4704-1a50-426c-999c-7d2dfa846a3e)

Reply:
ok, updated to:
```ts
installWrangler: z
	.union([z.null().transform(() => "beta"), z.string()])
	.optional(),
```

so that:
 - it is optional and we get `undefined` if not provided
 - with `--install-wrangler` we get `'beta'`
 - with `--install-wrangler=X` we get `X`
 
And typescript's happy:
![Screenshot 2023-08-11 at 17 20 17](https://github.com/GregBrimble/pages-e2e-tests/assets/61631103/ca5eada6-5a86-45a3-aba0-61921427a4b7)

(although `string | undefined` would be more correct 🤷)

Reply:
Ah perfect. You could try also adding a z.undefined() back into the union now you've made it optional? That might fix the typing issue.

Reply:
> Ah perfect. You could try also adding a z.undefined() back into the union now you've made it optional? That might fix the typing issue.

no, having both `z.null()` and `z.undefined()` in the union causes the `installWrangler` type to be `never` 🤷

Reply:
zod seems to be fine with it, I am not sure if it could be an issue with zodcli

see:
![Screenshot 2023-08-11 at 19 02 25](https://github.com/GregBrimble/pages-e2e-tests/assets/61631103/5ba9ed61-cd5f-4456-8e2d-69703d112854)

(https://stackblitz.com/edit/typescript-tjajki?file=index.ts)